src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -166,8 +166,8 @@ std::string GcsActor::GetRayNamespace() const {
 }
 
 TaskSpecification GcsActor::GetCreationTaskSpecification() const {
-  const auto &task_spec = actor_table_data_.task_spec();
-  return TaskSpecification(task_spec);
+  RAY_CHECK(task_spec_);
```

Comment:
Can you change this to `RAY_CHECK(state != DEAD)` and write a comment that task spec is not available when the actor is dead?

Reply:
done